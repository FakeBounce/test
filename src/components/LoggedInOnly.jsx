import React, { Component } from 'react';
import { Route, Redirect } from 'react-router-dom';
import AuthService from "services/AuthService";
import { omit } from 'lodash';

export default class LoggedInOnly extends Component {
  constructor(props) {
    super(props);
    this.onRouteRender= this.onRouteRender.bind(this);
  }

  onRouteRender(props) {
    const { component } = this.props;
    const PassedComponent = component;
    if (AuthService.loggedIn()) {
      return <PassedComponent {...props}/>;
    }
    return <Redirect to="/" />;
  }

  render() {
    const props = omit(this.props, ['component']);
    return <Route {...props} render={this.onRouteRender} />;
  }
}
