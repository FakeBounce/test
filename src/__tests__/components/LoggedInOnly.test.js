import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import LoggedInOnly from "components/LoggedInOnly";

configure({ adapter: new Adapter() });

describe("Testing LoggedInOnly component", () => {
  it("renders as expected", () => {
    const wrapper = shallow(<LoggedInOnly />);
    expect(wrapper).toMatchSnapshot();
  });
});
