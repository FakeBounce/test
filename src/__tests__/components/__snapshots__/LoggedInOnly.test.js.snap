// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoggedInOnly component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoggedInOnly />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
